pkg/argocd/app-controller.go

```diff
@@ -22,11 +22,13 @@ func createApplicationControllerStatefulSet(ctx context.Context, clientset *kube
 	for k, v := range argoLabels {
 		labels[k] = v
 	}
+	annotations := argoAnnotations
```

Comment:
Is the underlying map modified by the client? The labels above are copied so i'm not sure here if that is also needed.

Reply:
As far as I could see, the labels are copied because there are some additional labels being inserted as literals (line 18ff) and then the `argoLabels` map is effectively merged into that. I'm not doing that with the annotations, so I figured the copying wouldn't be necessary. I don't see why the client would modify the map.

Reply:
In Go slices and maps are passed by reference, unlike the other types. and i do wonder what the client does with the reference